.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: Test composite action
+on:
+  pull_request:
+    paths:
+      - 'action.yml'
+  workflow_dispatch:    
+
+jobs:
+  test-composite-action:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Echo kernelrelease
+        run: uname -a
+    
+      - name: Test action
+        uses: ./
+        with:
+          cache: 'true'
+          
+      - name: Check kernel sources
+        run: ls -la /usr/src
+  
+  test-composite-action-cached:
+    runs-on: ubuntu-latest
+    needs: test-composite-action
+    steps:
+      - name: Echo kernelrelease
+        run: uname -a
+        
+      - name: Store KERNEL_RELEASE
+        shell: bash
+        run: |
+          echo "KERNEL_RELEASE=$(uname -r)" >> $GITHUB_ENV  
+        
+      - name: Check kernel sources (empty)
+        run: ls -la /usr/src
+    
+      - name: Retrieve cached kernel sources
```

Comment:
This test is now smarter, since it will actually test that we are able to recover the cached sources.